### PR TITLE
Adding attachment to resources

### PR DESCRIPTION
**Pull request by Rupesh87**

### CODE REVIEW

File: couchdb-setup.sh [hunk 1]
```diff
@@ -47,6 +47,24 @@ else
...
+import urllib, json, sys, subprocess
+data = json.load(open('./design/resources/attachment.json'))
+for key in data:
+ id=key['doc_id']
+ image=key['filename']
+ command = 'http://a:a@localhost:2200/resources/'+id
+ response = urllib.urlopen(command)
+ jsondata = json.loads(response.read())
+ rev=jsondata['_rev']
+ putAttachment='curl -v -X PUT http://a:a@localhost:2200/resources/'+id+'/'+image+'?rev='+rev+' --data-binary @'+image+' -H Content-Type:image/jpg'
```
Comment (lmmrssa) [inline couchdb-setup.sh]: content-type needs to be dynamic
url of couch should use the constant variable COUCHURL

File: couchdb-setup.sh [hunk 2]
```diff
@@ -47,6 +47,24 @@ else
   COUCHURL=http://$COUCHUSER:$COUCHPASSWORD@$HOST:$PORT
 fi
 
+# Adding attachment to resource
+insert_attachment() {
```
Comment (lmmrssa) [inline couchdb-setup.sh]: make function dynamic so that it can take any json file or attach to any db

File: couchdb-setup.sh [hunk 3]
```diff
@@ -47,6 +47,24 @@ else
...
+import urllib, json, sys, subprocess
+data = json.load(open('./design/resources/attachment.json'))
+for key in data:
+ id=key['doc_id']
+ image=key['filename']
+ command = '$COUCHURL/resources/'+id
+ response = urllib.urlopen(command)
+ jsondata = json.loads(response.read())
+ rev=jsondata['_rev']
+ putAttachment='curl -v -X PUT $COUCHURL/resources/'+id+'/'+image+'?rev='+rev+' --data-binary @'+image+' -H Content-Type:image/jpg'
```
Comment (paulbert) [inline couchdb-setup.sh]: Let's also make `Content-Type` dynamic so we can add different kinds of files as well

File: design/resources/resources-mockup.json [hunk 4]
```diff
@@ -30,6 +31,7 @@
             "averageRating": ""
```
Comment (nirojdyola) [inline design/resources/resources-mockup.json]: averageRating is placed null in all section. We can rate it.

**Comment by lmmrssa** [comment]
looks good but still not sure about location of files. Its uploaded in assets right now.

File: design/resources/resources-mockup.json [hunk 5]
```diff
@@ -30,6 +31,7 @@
...
          },
          {
+            "_id":"0216c7208a72e79b29997d0fd400beac",
```
Comment (Pr0chin) [inline design/resources/resources-mockup.json]: IDs are same in all json doc..I think it should be changed.. or not

Comment (lmmrssa) [reply design/resources/resources-mockup.json]: ids are not same they look similar but definitely they are different

**Comment by Pr0chin** [comment]
What if we add some audio video attachment for resources ??? @rupesh

**Comment by Rupesh87** [comment]
@razu9861 please review carefully it had been already done

**Comment by Rupesh87** [comment]
@Pr0chin video and audio files are little bit large.

**Pull request merged**